OpenQA/Isotovideo/Utils.pm

```diff
@@ -12,15 +12,17 @@ use bmwqemu;
 use autotest;
 use Try::Tiny;
 
-our @EXPORT_OK = qw(checkout_git_repo_and_branch checkout_git_refspec
-  handle_generated_assets load_test_schedule);
+our @EXPORT_OK = qw(git_rev_parse checkout_git_repo_and_branch
+  checkout_git_refspec handle_generated_assets load_test_schedule);
+
+sub git_rev_parse ($dirname) {
+    my $version = (-e "$dirname/.git" ? qx{(git config --global --get safe.directory | grep -q $dirname || HOME=\$\(mktemp -d\) && mkdir -p \$HOME && git config --global --add safe.directory $dirname) && git -C $dirname rev-parse HEAD} : '') || 'UNKNOWN';
```

Comment:
`mktemp` should already create this directory, so why `mkdir`?

Also I would remove the directory afterwards.

Another thing I consider good practice is to use a template for tempdirs, so if for some reason a lot of them are there, we know at least where they come from.

Also for my taste the line is too long. The backslash stuff can be reduced by doing something like:
```
my $addsafe = q{HOME=$(mktemp -d) && git config --global --add safe.directory};
... qx{git config --global --get safe.directory | grep -q $dirname || $addsafe $dirname ) ... } ...
```

Reply:
Done together with a template for the temp dir.